### ISSUE TITLE
Unblessed reference error at doxygraph/doxygraph line 21, <DATA> line 2104.

### ISSUE THREAD

**Issue by rjeye** [jitsuCM/doxygraph#7]
So I'm trying to generate some class diagrams using the steps mentioned in the README. After installing all the missing modules it finally stops at an unblessed  reference in  doxygraph. Any fix for this ?

**Comment by qiuzhichang** [comment]
i comment the line of doxygraph/doxygraph:
 #$dot->foreach(sub { print $fh $_ });  

and it's works

**Comment by rollysys** [comment]
me too...

**Comment by Germwalker** [comment]
Same error ...
I know nothing about perl but i'll give it a try soon.

**Comment by Leprawel** [comment]
> Same error ... I know nothing about perl but i'll give it a try soon.

Any progress? Happened to me too.

**Comment by Leprawel** [comment]
According to information I got from perl matrix chat the possible cause of this issue is:

_The script uses internal methods of the GraphViz2 module. those internals changed between between GraphViz2 2.50 and 2.51, which broke the script. Specifically, the command attribute was changed from a Set::Array instance to a plain array, which is why you no longer can call methods on it._

**Comment by AndreiCostinescu** [comment]
> According to information I got from perl matrix chat the possible cause of this issue is:
> 
> _The script uses internal methods of the GraphViz2 module. those internals changed between between GraphViz2 2.50 and 2.51, which broke the script. Specifically, the command attribute was changed from a Set::Array instance to a plain array, which is why you no longer can call methods on it._

Yes, downgrading (or installing) the GraphViz2 perl module of version 2.50 fixes this issue (`cpanm GraphViz2@2.50`)

**Comment by Leprawel** [comment]
> > According to information I got from perl matrix chat the possible cause of this issue is:
> > _The script uses internal methods of the GraphViz2 module. those internals changed between between GraphViz2 2.50 and 2.51, which broke the script. Specifically, the command attribute was changed from a Set::Array instance to a plain array, which is why you no longer can call methods on it._
> 
> Yes, downgrading (or installing) the GraphViz2 perl module of version 2.50 fixes this issue (`cpanm GraphViz2@2.50`)

Thanks, managed to get it running. I also had to set `security.fileuri.strict_origin_policy = false` in my browsers `about:config`, because otherwise the graph.dot would not be allowed to load.
